src/main/java/org/jabref/gui/fieldeditors/GroupEditor.java

```diff
@@ -0,0 +1,56 @@
+package org.jabref.gui.fieldeditors;
+
+import java.util.List;
+import java.util.Optional;
+
+import javafx.scene.input.TransferMode;
+
+import org.jabref.gui.DragAndDropDataFormats;
+import org.jabref.gui.autocompleter.SuggestionProvider;
+import org.jabref.logic.integrity.FieldCheckers;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.Field;
+import org.jabref.model.entry.field.StandardField;
+import org.jabref.preferences.PreferencesService;
+
+public class GroupEditor extends SimpleEditor {
+
+    private Optional<BibEntry> bibEntry;
+
+    public GroupEditor(final Field field,
+                       final SuggestionProvider<?> suggestionProvider,
+                       final FieldCheckers fieldCheckers,
+                       final PreferencesService preferences,
+                       final boolean isMultiLine) {
+        super(field, suggestionProvider, fieldCheckers, preferences, isMultiLine);
+
+        this.setOnDragOver(event -> {
+            if (event.getDragboard().hasContent(DragAndDropDataFormats.GROUP)) {
+                event.acceptTransferModes(TransferMode.MOVE);
+            }
+            event.consume();
+        });
+
+        this.setOnDragDropped(event -> {
+            bibEntry.ifPresentOrElse((entry) -> {
+                if (event.getDragboard().hasContent(DragAndDropDataFormats.GROUP)) {
+                    List<String> group = (List<String>) event.getDragboard().getContent(DragAndDropDataFormats.GROUP);
+                    String changedGroup = entry.getField(StandardField.GROUPS)
+                                               .map(setGroup -> setGroup + (preferences.getBibEntryPreferences().getKeywordSeparator()) + (group.get(0)))
```

Comment:
Even after you revert these changes, I'd suggest changing some names to clarify,

- `List<String> group` -> `List<String> draggedGroups` (or `draggedGroup` if you add `.get(0)` to event.getDragboard().getContent(DragAndDropDataFormats.GROUP))
- `setGroup` -> `oldGroups`

Or if you believe something else is better

Reply:
> Here I can mentally deal with the `if (event.getDragboard().hasContent(DragAndDropDataFormats.GROUP))` before I start checking on what the `bibEntry` code does. Imo it makes it easier to see if there is a missing `event.setDropCompleted(false)` (line 43 😉)
> 
> ```java
>           if (event.getDragboard().hasContent(DragAndDropDataFormats.GROUP)) {
>                 final String group = (List<String>) event.getDragboard().getContent(DragAndDropDataFormats.GROUP).get(0);
>                 bibEntry.ifPresentOrElse(entry -> {
>                     final String changedGroup = entry.getField(StandardField.GROUPS)
>                                                      .map(setGroup -> setGroup + (preferences.getBibEntryPreferences().getKeywordSeparator()) + (group)).orElse(group);
>                     entry.setField(StandardField.GROUPS, changedGroup);
>                     event.setDropCompleted(true);
>                 }, () -> event.setDropCompleted(false));
>             } else {
>                 event.setDropCompleted(false);
>             }
>             event.consume();
>         });
> ```

Thank you for your explanation and it is inspiring. I think if-else is better with longer code, and people don't have to search the whole way down to find where 'success' is changed. I will keep it in mind when dealing with similar issues in the future😁. I have changed the name of the variables in my newest commit.